pandas/core/groupby/groupby.py

```diff
@@ -1416,15 +1416,18 @@ def _obj_1d_constructor(self) -> Type[Series]:
         return self.obj._constructor
 
     @final
-    def _bool_agg(self, val_test, skipna):
+    def _bool_agg(self, val_test, skipna: bool):
         """
         Shared func to call any / all Cython GroupBy implementations.
         """
 
-        def objs_to_bool(vals: np.ndarray) -> Tuple[np.ndarray, Type]:
+        def objs_to_bool(vals: ArrayLike) -> Tuple[np.ndarray, Type]:
             if is_object_dtype(vals):
                 vals = np.array([bool(x) for x in vals])
             else:
+                if isinstance(vals, ExtensionArray):
+                    vals = vals.to_numpy(dtype=bool, na_value=np.nan)
```

Comment:
in these cases we may want to call the EA method rather than cast to numpy, cc @jorisvandenbossche ?

Reply:
Yep not sure this is the best approach, was just looking at how masked groupby quantile was handled (IIUC the only masked groupby algo, along with this one). Want to start looking into converting cython algos to better handle EA through taking masks, any advice on a good approach here and elsewhere would be much appreciated.

What do you mean by call the EA method? Won't we eventually need to end up as an ndarray for cython algo?

Reply:
also pls use elif

Reply:
done